packages/react-router/lib/dom/ssr/single-fetch.tsx

```diff
@@ -412,6 +412,9 @@ export function singleFetchUrl(reqUrl: URL | string) {
 
 async function fetchAndDecode(url: URL, init: RequestInit) {
   let res = await fetch(url, init);
+  if (res.status === 404) {
+    throw new ErrorResponseImpl(404, "Not Found", true);
+  }
```

Comment:
Bubble a proper 404 to our UI if we're in a prerendered app and there is no static file on the server